src/utils/iqeEnablement.ts

```diff
@@ -13,14 +13,18 @@ let fetchResults: Record<string, unknown> = {};
 
 const DENIED_CROSS_CHECK = 'Access denied from RBAC on cross-access check';
 const AUTH_ALLOWED_ORIGINS = [location.origin, 'https://api.openshift.com', 'https://api.stage.openshift.com'];
+const AUTH_EXCLUDED_URLS = ['https://api.openshift.com/api/upgrades_info/', 'https://api.stage.openshift.com/api/upgrades_info/'];
 
-const checkOrigin = (path: URL | Request | string = '') => {
+const shouldInjectAuthHeaders = (path: URL | Request | string = '') => {
   if (path instanceof URL) {
-    return AUTH_ALLOWED_ORIGINS.includes(path.origin);
+    return AUTH_ALLOWED_ORIGINS.includes(path.origin) && !AUTH_EXCLUDED_URLS.some((url) => path.href.includes(url));
   } else if (path instanceof Request) {
-    return AUTH_ALLOWED_ORIGINS.some((origin) => path.url.includes(origin));
+    return AUTH_ALLOWED_ORIGINS.some((origin) => path.url.includes(origin)) && !AUTH_EXCLUDED_URLS.some((url) => path.url.includes(url));
```

Comment:
@Hyperkid123 since we are adding this line in 3 sports - I wonder if for maintainability it would be best to create a helper function for all the duplication:
```ts
const someURL = (
  base: [string],
  target: {
    includes: (searchElement: string) => boolean;
  }
) => base.some((url) => target.includes(url));

return (
  someURL(AUTH_ALLOWED_ORIGINS, path.url) &&
  !someURL(AUTH_EXCLUDED_URLS, path.url)
);
```

Reply:
I'll come up with something nice.